app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsFormModal/LocalUnitsForm/index.tsx

```diff
@@ -509,7 +571,9 @@ function LocalUnitsForm(props: Props) {
 
     return (
         <div className={styles.localUnitsForm}>
-            {readOnly && isDefined(actionsContainerRef.current) && (
+            {!localUnitDetailsResponse?.is_locked
+                && readOnlyFromProps
+                && isDefined(actionsContainerRef.current) && (
```

Comment:
It's not correct to check for ref on render. Change in ref will not be updated on render.

Reply:
Please ask @frozenhelium about this logic, i have not made any changes regarding check for ref on render.